lightning/checkpoints/checkpoints.go

```diff
@@ -937,57 +936,68 @@ func (*NullCheckpointsDB) DumpChunks(context.Context, io.Writer) error {
 }
 
 func (cpdb *MySQLCheckpointsDB) RemoveCheckpoint(ctx context.Context, tableName string) error {
-	var (
-		deleteChunkFmt  string
-		deleteEngineFmt string
-		deleteTableFmt  string
-		arg             interface{}
-	)
-
-	if tableName == "all" {
-		deleteChunkFmt = "DELETE FROM %[1]s.%[2]s WHERE table_name IN (SELECT table_name FROM %[1]s.%[3]s WHERE node_id = ?)"
-		deleteEngineFmt = "DELETE FROM %[1]s.%[2]s WHERE table_name IN (SELECT table_name FROM %[1]s.%[3]s WHERE node_id = ?)"
-		deleteTableFmt = "DELETE FROM %s.%s WHERE node_id = ?"
-		arg = nodeID
-	} else {
-		deleteChunkFmt = "DELETE FROM %s.%s WHERE table_name = ?%.0s" // the %.0s is to consume the third parameter.
-		deleteEngineFmt = "DELETE FROM %s.%s WHERE table_name = ?%.0s"
-		deleteTableFmt = "DELETE FROM %s.%s WHERE table_name = ?"
-		arg = tableName
-	}
-
-	deleteChunkQuery := fmt.Sprintf(deleteChunkFmt, cpdb.schema, checkpointTableNameChunk, checkpointTableNameTable)
-	deleteEngineQuery := fmt.Sprintf(deleteEngineFmt, cpdb.schema, checkpointTableNameEngine, checkpointTableNameTable)
-	deleteTableQuery := fmt.Sprintf(deleteTableFmt, cpdb.schema, checkpointTableNameTable)
 	s := common.SQLWithRetry{
 		DB:     cpdb.db,
 		Logger: log.With(zap.String("table", tableName)),
 	}
-	err := s.Transact(ctx, "remove checkpoints", func(c context.Context, tx *sql.Tx) error {
-		if _, e := tx.ExecContext(c, deleteChunkQuery, arg); e != nil {
+
+	if tableName == "all" {
+		return s.Exec(ctx, "remove all checkpoints", "DROP SCHEMA "+cpdb.schema)
+	}
+
+	deleteChunkQuery := fmt.Sprintf("DELETE FROM %s.%s WHERE table_name = ?", cpdb.schema, checkpointTableNameChunk)
+	deleteEngineQuery := fmt.Sprintf("DELETE FROM %s.%s WHERE table_name = ?", cpdb.schema, checkpointTableNameEngine)
+	deleteTableQuery := fmt.Sprintf("DELETE FROM %s.%s WHERE table_name = ?", cpdb.schema, checkpointTableNameTable)
+
+	return s.Transact(ctx, "remove checkpoints", func(c context.Context, tx *sql.Tx) error {
+		if _, e := tx.ExecContext(c, deleteChunkQuery, tableName); e != nil {
 			return errors.Trace(e)
 		}
-		if _, e := tx.ExecContext(c, deleteEngineQuery, arg); e != nil {
+		if _, e := tx.ExecContext(c, deleteEngineQuery, tableName); e != nil {
 			return errors.Trace(e)
 		}
-		if _, e := tx.ExecContext(c, deleteTableQuery, arg); e != nil {
+		if _, e := tx.ExecContext(c, deleteTableQuery, tableName); e != nil {
 			return errors.Trace(e)
 		}
 		return nil
 	})
-	return errors.Trace(err)
+}
+
+func (cpdb *MySQLCheckpointsDB) MoveCheckpoints(ctx context.Context, taskID int64) error {
+	newSchema := fmt.Sprintf("`%s.%d.bak`", cpdb.schema[1:len(cpdb.schema)-1], taskID)
```

Comment:
is it better to use a function `unescapedSchemaName `?

Reply:
Added a comment explaining why unescaping isn't used (to keep the escaped `` ` ``s)